txpool/txpool.go

```diff
@@ -284,6 +290,10 @@ func (p *TxPool) Drop() {
 		"nonce", tx.Nonce,
 		"addr", tx.From.String(),
 	)
+	go p.eventManager.fireEvent(&proto.TxPoolEvent{
```

Comment:
I would not force the user of the ```fireEvent``` api (txpool) to use ```go``` everytime. Do the async part inside the fireEvent itself.

Reply:
You're right, the caller shouldn't need to know it needs to run the `fireEvent` asynchronously 👍 

Fixed:

b71384a